File: src/main/java/src/SistemaRPG/GestorBatalla.java
package SistemaRPG;

import java.util.ArrayList;
import java.util.List;

public class GestorBatalla {
    private List<Personaje> personajes;

    public GestorBatalla() {
        personajes = new ArrayList<>();
    }

    public void agregarPersonaje(Personaje personaje) {
        personajes.add(personaje);
        System.out.println(personaje.getNombre() + " acaba de spawnear.");
    }

    public void accion(Personaje atacante, Personaje objetivo) {
        atacante.atacar(objetivo);
    }

    public void mostrarEstado() {
        System.out.println("Estado:");
        for (Personaje personaje : personajes) {
            System.out.println(personaje);
        }
    }
}

File: src/main/kotlin/SistemaRPG/GestorBatalla.kt
package SistemaRPG

class GestorBatalla {
    private val personajes: MutableList<Personaje> = ArrayList()

    fun agregarPersonaje(personaje: Personaje) {
        personajes.add(personaje)
        println(personaje.nombre + " acaba de spawnear.")
    }

    fun accion(atacante: Personaje, objetivo: Personaje?) {
        atacante.atacar(objetivo)
    }

    fun mostrarEstado() {
        println("Estado:")
        for (personaje in personajes) {
            println(personaje)
        }
    }
}